From 59f9c84f2e207d47704a449a473a0091c311e182 Mon Sep 17 00:00:00 2001
From: Pinak17 <pinakroychowdhury@gmail.com>
Date: Fri, 11 Aug 2017 22:51:41 +0530
Subject: [PATCH] shortlog: Optimization of strlen.

In the function 'urldecode.c' the function strlen has been called more
than once over same string. I have created a temporary variable named
'lenOfString' to store the length of string.

fixes #1

Signed-off-by: Pinak17 <pinakroychowdhury@gmail.com>
---
 src/urldecode.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/urldecode.c b/src/urldecode.c
index d6bb8f1..f2ed7c4 100644
--- a/src/urldecode.c
+++ b/src/urldecode.c
@@ -36,12 +36,13 @@ char* urldecode(const char* s)
     if (s == NULL)    return NULL;
 
     // allocate enough (zeroed) memory for an undecoded copy of s
-    char* t = calloc(strlen(s) + 1, 1);
+	int lenOfString = strlen(s);
+    char* t = calloc(lenOfString + 1, 1);
     if (t == NULL)    return NULL;
 
     // iterate over characters in s, decoding percent-encoded octets, per
     // https://www.ietf.org/rfc/rfc3986.txt
-    for (int i = 0, j = 0, n = strlen(s); i < n; i++, j++) {
+    for (int i = 0, j = 0, n = lenOfString; i < n; i++, j++) {
         if (i < n - 2 && s[i] == '%') {
             char octet[3];
             octet[0] = s[i + 1];
-- 
2.13.3.windows.1

